### PR TITLE
docs: break testing doc into multiple docs

### PR DESCRIPTION
**Pull request by kapunahelewong**
This commit breaks up the testing document into nine total documents, with each document focusing on an existing section of the current testing documentation. There are no content changes. The only copy changes are those that reference other parts of the former testing guide. So, for example if a sentence previously referred to the section above but that section is now in a different doc the reference now goes to that document and the wording is changed appropriately.

This PR only breaks the document apart, and does not rewrite the content. The content is as close as possible to the original.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Now there are new documents where the original testing document used to be. The new documents are each one of the major sections from the original testing.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

https://pr37957-ca2700e.ngbuilds.io/guide/testing

### CODE REVIEW

File: aio/content/navigation.json [hunk 1]
```diff
@@ -536,19 +536,65 @@
             "url": "guide/template-typecheck",
             "title": "Template Type-checking",
             "tooltip": "Template type-checking in Angular."
-         }
-        ]
-      },
+           }
+         ]
+        },
```
Comment (gkalpak) [inline aio/content/navigation.json:541]: The indentation seems to be incorrect here (and below).

File: aio/content/guide/upgrade-setup.md [hunk 2]
```diff
@@ -308,7 +308,7 @@ So when IE is refreshed (manually or automatically by `ng serve`), sometimes the
 
 ## Appendix: Test using `fakeAsync()/async()`
 
-If you use the `fakeAsync()/async()` helper function to run unit tests (for details, read the [Testing guide](guide/testing#async-test-with-fakeasync)), you need to import `zone.js/dist/zone-testing` in your test setup file.
+If you use the `fakeAsync()/async()` helper function to run unit tests (for details, read the [Testing guide](guide/testing-components-scenarios#async-test-with-fakeasync)), you need to import `zone.js/dist/zone-testing` in your test setup file.
```
Comment (gkalpak) [inline aio/content/guide/upgrade-setup.md:311]:
```suggestion
If you use the `fakeAsync()/async()` helper function to run unit tests (for details, read the [Testing guide](guide/testing-components-scenarios#fake-async)), you need to import `zone.js/dist/zone-testing` in your test setup file.
```

**Review by gkalpak — APPROVED** [review]
LGTM for `fw-upgrade` (with one minor comment) :+1:

Reviewed-for: fw-upgrade

**Review by pkozlowski-opensource — APPROVED** [review]

**Review by aikithoughts — APPROVED** [review]
LGTM. Thank you for the excellent commit message!

Reviewed-for: global-docs-approvers

**Review by aikithoughts — APPROVED** [review]

**Review by jelbourn — APPROVED** [review]
LGTM, using global approvers for codeowners

Reviewed-for: global-approvers

**Pull request closed without merging**